packages/api/src/hooks/p2p/useAdvertUpdate.ts

```diff
@@ -0,0 +1,54 @@
+import { useCallback, useMemo } from 'react';
+import useMutation from '../../useMutation';
+import useInvalidateQuery from '../../useInvalidateQuery';
+
+type TPayload = Parameters<ReturnType<typeof useMutation<'p2p_advert_update'>>['mutate']>[0]['payload'];
+
+/** A custom hook that updates a P2P advert. This can only be used by an approved P2P advertiser.
+ * 
+ * To update an advert, specify the payload arguments that should be updated, for instance:
+ * @example
+ *  mutate({
+        "id": 1234, // required
+        "is_active": 0 // optional
+    });
+ * 
+*/
+const useAdvertUpdate = () => {
+    const invalidate = useInvalidateQuery();
+    const {
+        data,
+        mutate: _mutate,
+        ...rest
+    } = useMutation('p2p_advert_update', {
+        onSuccess: () => {
+            invalidate('p2p_advert_list');
+        },
+    });
+
+    const mutate = useCallback((payload: TPayload) => _mutate({ payload }), [_mutate]);
```

Comment:
should we not split this into two hooks, one for updating, the other for deleting? similar to niloofar's PR: https://github.com/binary-com/deriv-app/pull/12168/files

Reply:
Oh shee I didn't see the `deleted` property 🤣 
I'll separate it into another hook